### PR TITLE
Translate API

### PR DESCRIPTION
**Pull request by mau-fyi**
Adds buttons capable of switching the pages language.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java:39]: It would be good to validate the input language to make sure it is a language you support.
I was able to change the associated language from the console. So something like the code below would be nice. It's always good to validate input. That way you can also send back errors to the client (you don't need to do that now).

public class TranslateAPIServelet ... {
 private static Set<String> SUPPORTED_LANGUAGES = new HashSet<>(Arrays.asList("en", "es", "fr"));

public void doPost(...) {
  if (validateInput(request)) {
    // translate
  }
  // dont translate


}

Checking for language strings should be ok for now. But you could also use the java locale library - https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html

**Review by sbahaud — COMMENTED** [review]
nit: Please update PR description to something like "Add buttons to the main page to Translate page using Translate API"

I left a few comments.

File: portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java
Comment (mau-fyi) [reply portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java:39]: Good idea. I remember when I misspelled a language code my entire page was replaced with html responses.
As for the local, I'd love to look into that once I finish up with the other library.

File: portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java]: This should be private static.
In java consts are stored as static, since they don't need to change. Also private, since it does not have to be accessed from outside this class.
private static class level const vars are always UPPER_CASE. So
private static Set<String> ACCEPTED_LANGUAGES = ...

File: portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java]: I'll accept this for now. But would be good to put this in the constructor instead of in the doPost.

File: portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java
Comment (sbahaud) [reply portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java:39]: HAving a separate validate method, makes the code more readable. But ok for now.

**Review by sbahaud — APPROVED** [review]
Approving with one required fix. Please make that fix before submitting.

File: portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java
Comment (mau-fyi) [reply portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java]: Got it!

File: portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java
Comment (mau-fyi) [reply portfolio/src/main/java/com/google/sps/servlets/TranslateAPIServlet.java]: Ohhh thank you! I was wondering how I could have it be instantiated when the class is used. I'll make a commit for this after merging!

**Pull request merged**
